online-shop/src/components/products/ProductDetailsComponent.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { SyntheticEvent } from 'react';
+
+import './ProductDetailsComponent.scss';
+
+import { BACKEND_API, PRODUCTS_PATH } from '../../constants';
+import { ControlButton } from '../utils/ControlButton';
+import { IconControlButton } from '../utils/IconControlButton';
+import { IProduct } from '../../types';
+
+interface IProps {
+    id: any,
+    history: any,
+    onAddToCartClick: (product: IProduct) => void,
+    onDeleteProductClick: (id: number) => void
+}
+
+interface IState {
+    product: IProduct,
+    isFetching: boolean,
+    error: any,
+}
+
+export default class ProductDetails extends React.Component<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+        this.state = {
+            product: {} as any,
+            isFetching: true,
+            error: null,
+        };
+    }
+
+    componentDidMount() {
+        fetch(`${BACKEND_API}${PRODUCTS_PATH}/${this.props.id}`)
+            .then(response => response.json())
+            .then(result => {
+                this.setState({ product: result, isFetching: false })
+            })
+            .catch(error => this.setState({ error, isFetching: false }));
+    }
+
+    render() {
+        const { product, isFetching, error } = this.state;
+        if (error) {
+            return <p>{error.message}</p>
+        }
+
+        if (isFetching) {
+            return <p>Fetching data ....</p>
+        }
+
+        return (
+            <section className='section'>
+                <nav className="navbar is-transparent">
+                    <div className='navbar-start'>
+                        <p className='subtitle is-5'>
+                            <strong>Product: {product.name}</strong>
+                        </p>
+                    </div>
+                    <div className="navbar-end">
+                        <div className="field is-grouped">
+                            <IconControlButton buttonName='is-success' buttonTitle='Add to shopping cart'
+                                iconTitle='fas fa-shopping-cart' clickEvent={(e: SyntheticEvent) => this.addToShoppingCart(product)} />
```

Comment:
You don't need to specify the `e: SyntheticEvent` parameter if you don't use it.